components/TradingCardGrid.jsx

```diff
@@ -1,10 +1,26 @@
+import { useFilterContext } from '../context/FilterContext';
 import getFullName from '../util/getFullName';
 import TradingCard from './TradingCard';
 
 export default function TradingCardGrid({ teamMembers }) {
+  const { project, roles, availability } = useFilterContext();
+
+  const filteredTeamMembers = teamMembers.filter((member) => {
+    const hasHoursAvailable = member.weeklyCapacity - member.forecastedHours > 0;
+
+    if (!hasHoursAvailable && availability === 'available') return false;
+    if (hasHoursAvailable && availability === 'unavailable') return false;
+
+    if (!roles.includes(member.jobTitle)) return false;
```

Comment:
Since "no roles selected" should behave the same way as if all roles are selected, we should update this accordingly.

```js
    if (roles.length && !roles.includes(member.jobTitle)) return false;
```